lib/src/model/comment_referable.dart

```diff
@@ -62,6 +76,50 @@ mixin CommentReferable implements Nameable {
     return result;
   }
 
+  /// Looks up references by [scope], skipping over results that do not match
+  /// the given filter.
+  ///
+  /// Override if [Scope.lookup] may return a [PrefixElement] or other elements
+  /// not corresponding to a [CommentReferable], but you still want to have
+  /// an implementation of [scope].
+  CommentReferable lookupViaScope(ReferenceChildrenLookup referenceLookup,
+      bool Function(CommentReferable) filter) {
+    var resultElement = scope.lookupPreferGetter(referenceLookup.lookup);
+    if (resultElement is PrefixElement) {
+      assert(false,
+          'PrefixElement detected, override [lookupViaScope] in subclass');
+      return null;
+    }
+    return recurseChildrenAndFilter(referenceLookup,
+        ModelElement.fromElement(resultElement, packageGraph), filter);
+  }
+
+  CommentReferable _lookupViaReferenceChildren(
+          ReferenceChildrenLookup referenceLookup,
+          bool Function(CommentReferable) filter) =>
+      recurseChildrenAndFilter(
+          referenceLookup, referenceChildren[referenceLookup.lookup], filter);
+
+  /// Assuming we found a [result], recurse through children, skipping over
+  /// results that do not match the filter.
+  CommentReferable recurseChildrenAndFilter(
+      ReferenceChildrenLookup referenceLookup,
+      CommentReferable result,
+      bool Function(CommentReferable) filter) {
+    assert(result != null);
+    if (referenceLookup.remaining.isNotEmpty) {
+      result = result?.referenceBy(referenceLookup.remaining,
```

Comment:
The `?.` is not necessary with the above assert, right?

Reply:
correct, fixed